section3/reflection.md

```diff
@@ -1,17 +1,37 @@
+- S.M.A.R.T goals is something I recently came across and have been trying to to improve.  In the past I always had issues with creating goals and when i did they were always too far ahead.  This led me to getting down on myself when I didn't achieve them in a timely fashion or the impending feeling of it being too far out o reach would discourage me.
+- This one always seems to be overused but it is the truth.  I learn a lot more from my mistakes rather than my successes.  Especially in education my mistakes or incorrect answers have always stayed with me longer than the times when I was correct right from the beginning.  It's important to fail in life it is how we truly learn.  There is also a lesson in what we do with that failure to make sure we succeed in the future.
 
 1. In which ways do you currently demonstrate a Growth Mindset? In which ways do you _not_?
+- I currently learn a lot from failure.  I also tend to always want to stretch my knowledge.  This suggests that I tend to be eager to learn new things or techniques.  
+- I don't show a growth mindset in the way when it comes to questions.  I don't tend to ask many questions even if I'm not 100% sure of what I'm doing.  I know it something I have to work on because I don't know all the answers but I rely on my ability to figure it out later.  This does not always turn out positive though.  
 
 1. What is a Hash, and how is it different from an Array?
+- A Hash allows you to store multiple different types of data.  You can have both integers and strings in a hash.  Whereas in an array they must all be the same type.
 
 1. In the space below, create a Hash stored to a variable named `pet_store`.  This hash should hold an inventory of items and the number of that item that you might find at a pet store.
 
+```
+pet_store = {
+  chew_toys => 1000,
+  treats => 3000,
+  leashes => 200,
+  brand_food => 'Pedigree'
+}
+
+```
+
 1. Given the following `states = {"CO" => "Colorado", "IA" => "Iowa", "OK" => "Oklahoma"}`, how would you access the value `"Iowa"`?
+- `p states["IA"]`
 
 1. With the same hash above, how would we get all the keys?  How about all the values?
-
+- all keys: `p states.keys`
+- all values: `p states.values
+`
 1. What is another example of when we might use a hash?  In your example, why is a hash better than an array?
+- An example would be profiles on a website.  A hash would be more according because of the need of different data types that are required in a profile.  They aren't all integers or all strings.  A hash would be more suited to fit everything you need into it.  
 
 1. What questions do you still have about hashes?
+- I'll do more research into this but I learned the pros towards using a hash over an array.  My question is, is the only benefit of an array over a has is that an array is ordered? It just kind of seems like why even use an array over a hash? This could just be from my lack of experience with hashes and arrays.  
```

Comment:
Arrays are typically used to hold a list of things
Hashes are usually used to hold object style information

As you progress in Turing, you will practice both often, so you will see real world applications of the differences.
But to see a preview, research JSON data. You will see large Hash sets that sometimes have arrays inside of them! This is common data used by BE (and sometimes FE)